thrift.js

```diff
@@ -178,6 +178,15 @@ Thrift.prototype.compileService = function compileService(def) {
     var self = this;
     var service = new ThriftService({strict: self.strict});
     service.compile(def, self);
+
+    if (def.baseService) {
+        var baseService = self.resolve(def.baseService);
+        for (var index = 0; index < baseService.functions.length; index++) {
+            var thriftFunction = baseService.functions[index];
+            service.addFunction(thriftFunction);
+        }
+    }
```

Comment:
This needs to be moved to the link step. `resolve` is not guaranteed to work during compile.